### PR TITLE
[mono][tests] Enable runtime tests for local aliasing on Mono Interpreter

### PR DESCRIPTION
**Pull request by kotlarmilos**
This PR should enable a subset of runtime tests added in https://github.com/dotnet/runtime/pull/74665 on Mono Interpreter. Interpreter doesn't create temporary instances when creating an object, so `ProblemWithCapturingCtors` test is disabled.

### CODE REVIEW

File: src/tests/JIT/Regression/JitBlue/Runtime_74635/Runtime_74635.ilproj [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.IL">
   <PropertyGroup>
     <Optimize>True</Optimize>
+    <DefineConstants Condition="'$(RuntimeVariant)' == 'monointerpreter'">MONO_INTERPRETER</DefineConstants>
```
Comment (SingleAccretion) [inline src/tests/JIT/Regression/JitBlue/Runtime_74635/Runtime_74635.ilproj:4]: I am surprised this works, given ILASM doesn't support specifying `#define`s on command line, does the IL SDK assist in adding the files with them?

Comment (kotlarmilos) [reply src/tests/JIT/Regression/JitBlue/Runtime_74635/Runtime_74635.ilproj:4]: Unfortunately it didn't work. Alternative option could be `ConditionalFact`, similar to the following.
```
    // [ConditionalFact(typeof(TestLibrary.PlatformDetection), nameof(TestLibrary.PlatformDetection.Is64BitProcess))]
    .custom instance void [Microsoft.DotNet.XUnitExtensions]Xunit.ConditionalFactAttribute::.ctor(class [System.Runtime]System.Type,
                                                                                                  string[]) = ( 01 00 61 54 65 73 74 4C 69 62 72 61 72 79 2E 50   // ..aTestLibrary.P
                                                                                                                6C 61 74 66 6F 72 6D 44 65 74 65 63 74 69 6F 6E   // latformDetection
                                                                                                                2C 20 54 65 73 74 4C 69 62 72 61 72 79 2C 20 56   // , TestLibrary, V
                                                                                                                65 72 73 69 6F 6E 3D 30 2E 30 2E 30 2E 30 2C 20   // ersion=0.0.0.0, 
                                                                                                                43 75 6C 74 75 72 65 3D 6E 65 75 74 72 61 6C 2C   // Culture=neutral,
                                                                                                                20 50 75 62 6C 69 63 4B 65 79 54 6F 6B 65 6E 3D   //  PublicKeyToken=
                                                                                                                6E 75 6C 6C 01 00 00 00 0E 49 73 36 34 42 69 74   // null.....Is64Bit
                                                                                                                50 72 6F 63 65 73 73 00 00 )                      // Process..
```

Comment (kotlarmilos) [reply src/tests/JIT/Regression/JitBlue/Runtime_74635/Runtime_74635.ilproj]: It seems that the `PlatformDetection` doesn't work as the test got executed. According to the logs, `DOTNET_RUNTIME_VARIANT` is set to `monointerpreter`. I will try to disable it through `issues.targets`.

**Comment by kotlarmilos** [comment]
The PR is ready for review. Test `ProblemWithCapturingCtors` is disabled on Interpreter, while others are enabled.

**Review by BrzVlad — APPROVED** [review]

**Pull request merged**